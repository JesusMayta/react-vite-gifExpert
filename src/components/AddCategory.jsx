import { useState } from "react";
import PropTypes from 'prop-types';

export const AddCategory = ({ setCategories, onNewCategory }) => {

    const [inputValue, setInputValue] = useState('');

    const handleChangeInput = (e) => {
        setInputValue(e.target.value);
    };

    const handleSubmit = (e) => {
        e.preventDefault();
        const category = inputValue.trim();
        if (category.length <= 1) return;
        // setCategories(categories => [inputValue, ...categories]);
        onNewCategory(category);
        setInputValue('');
    };

    return (
        <form onSubmit={handleSubmit} aria-label="form">
            <input
                type="text"
                placeholder="Buscar gifs"
                value={inputValue}
                onChange={handleChangeInput}
            />
        </form>
    );
};


AddCategory.propTypes = {
    onNewCategory: PropTypes.func.isRequired,
}; 