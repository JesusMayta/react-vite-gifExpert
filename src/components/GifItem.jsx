import PropTypes from 'prop-types';

export const GifItem = ({ title, url }) => {
    return (
        <div>
            <img width="300" height="400" src={url} alt={title} />
            <p>{title}</p>
        </div>
    );
};

// Proptypes
GifItem.propTypes = {
    title: PropTypes.string.isRequired,
    url: PropTypes.string.isRequired,
}