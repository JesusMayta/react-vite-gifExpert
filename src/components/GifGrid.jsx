import PropTypes from 'prop-types';

import { useFetchGifs } from "../hooks/useFetchGifs";
import { GifItem } from "./GifItem";

export const GifGrid = ({ category }) => {

    const { images, isLoading } = useFetchGifs(category);

    return (
        <>

            <h3>{category}</h3>
            {isLoading && (<h3>Cargando...</h3>)}
            {/* <button onClick={() => setCounter(counter + 1)}>+1</button>  */}
            <div className="card_grid" >
                {
                    images.map((image) => (
                        <GifItem key={image.id} {...image} />
                    ))
                }
            </div>

        </>
    );
};

GifGrid.propTypes = {
    category: PropTypes.string.isRequired,
};
