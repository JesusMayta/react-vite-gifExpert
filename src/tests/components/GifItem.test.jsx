import { render, screen } from "@testing-library/react";
import { GifItem } from "../../components/GifItem";

describe('Pruebas en <GifItem />', () => {

    const title = 'Saitama';
    const url = 'https://one-punch.com/saitama.jpg';

    test('Debe hacer match con el Snapshot', () => {

        const { container } = render(<GifItem title={title} url={url} />);

        expect(container).toMatchSnapshot();
    });

    test('Debe mostrar la imagen con el URL y el ALT indicado', () => {

        render(<GifItem title={title} url={url} />);

        const { src, alt } = screen.getByRole('img');
        expect(src).toBe(url);
        expect(alt).toBe(title);
    });

    test('Debe mostrar el titulo en el componente', () => {

        render(<GifItem title={title} url={url} />);
        expect(screen.getByText(title)).toBeTruthy();
    });

});
