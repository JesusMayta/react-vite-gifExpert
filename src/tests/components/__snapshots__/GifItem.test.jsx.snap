// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <GifItem /> Debe hacer match con el Snapshot 1`] = `
<div>
  <div>
    <img
      alt="Saitama"
      height="400"
      src="https://one-punch.com/saitama.jpg"
      width="300"
    />
    <p>
      Saitama
    </p>
  </div>
</div>
`;
